transcom/mymove
# INT-B-19980 Fixed GCC/Gross Incentive Storage SIT Calculations

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-19980)

## Summary

This branch fixes the way that SIT/Storage costs and receipts are factored into the GCC and Gross Incentive that are visible on the PPM Closeout review screen.

1. The GCC now uses the **estimated** SIT cost, this cost is calculated when the SC counsels a move and inputs an estimated SIT weight and date range.
2. The Gross Incentive now uses the **actual** SIT cost, which is gathered from the receipt entered by the customer during PPM Documentation Upload. Note that this will not occur if the SC rejects or excludes the related storage receipt when they are reviewing the PPM closeout.


### How to test

1. Create a PPM move as usual, be sure to check "Yes" when asked if you plan to use SIT. Sign as the customer.
2. As an SC, before you submit the move, click the "Edit Shipment" button, then near the bottom check "Yes" for "Does the customer plan to use SIT?". Then fill out the estimated weights/dates.
![Screenshot 2024-06-05 at 3 30 35 PM](https://github.com/transcom/mymove/assets/146959895/e5d75719-b204-4859-aaa8-7b651f120c6c)

3. When you click next, take note of the estimated amount that is given for SIT, shown in the screenshot below:
![Screenshot 2024-06-05 at 3 30 45 PM](https://github.com/transcom/mymove/assets/146959895/7976133e-83b6-407d-95ec-2f5f8df2586b)

4. Submit move, then as the customer upload the documentation. Be sure to add a "Storage" expense receipt, and make the amount different than that shown for the "estimate" above.
5. Sign as customer, then as SC access the move from the PPM closeout queue. Accept the storage receipt.
6. At the review screen, you should now see that the GCC and the Gross Incentive are different based on the difference in amounts between the expected and actual storage costs. For example, in this screenshot, my estimated expenses were $1416, but the actual receipt listed $3,111, leading to a difference of ~$1,695 between the two:
![Screenshot 2024-06-05 at 3 33 02 PM](https://github.com/transcom/mymove/assets/146959895/72006d71-27a5-47db-ad08-d9aa418f11a5)

7. Now before submitting, go back and reject the storage receipt, you should now see the Gross Incentive drop lower by that receipt's amount, for example, this is my screen after I reject the previous storage receipt:
![Screenshot 2024-06-05 at 3 35 19 PM](https://github.com/transcom/mymove/assets/146959895/c0fe42ce-9c81-4839-ab0f-226519aeab0f)
